library/java/src/io/envoyproxy/envoymobile/engine/types/EnvoyStringAccessor.java

```diff
@@ -0,0 +1,11 @@
+package io.envoyproxy.envoymobile.engine.types;
+
+import java.nio.ByteBuffer;
+
+public interface EnvoyStringAccessor {
+
+  /**
+   * Called to retrieve a string from the Application
+   */
+  ByteBuffer getString();
```

Comment:
decided to return a ByteBuffer here rather than a String to take advantage of the utilities we already have that bridge it into an envoy_data. 

Although it wouldn't be too complicated to create a bridging utility from String to envoy_data if we'd rather use that type.

Reply:
Seems fine. Don't we already have a utility for String though?

Reply:
Not afaict? We have a byte buffer and a byte array utility. I guess the byte array one can be used with a jstring? But you have to copy.